MUONdigi/src/MUONsimpleDigitizer.cpp

```diff
@@ -0,0 +1,183 @@
+
+StatusCode MUONsimpleDigitizer::initialize() {
+  // Initialize random services
+  if (service("RndmGenSvc", m_randSvc).isFailure()) {
+    error() << "Couldn't get RndmGenSvc!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_gauss_x.initialize(m_randSvc, Rndm::Gauss(0., m_x_resolution)).isFailure()) {
+    error() << "Couldn't initialize RndmGenSvc!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_gauss_y.initialize(m_randSvc, Rndm::Gauss(0., m_y_resolution)).isFailure()) {
+    error() << "Couldn't initialize RndmGenSvc!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_gauss_z.initialize(m_randSvc, Rndm::Gauss(0., m_z_resolution)).isFailure()) {
+    error() << "Couldn't initialize RndmGenSvc!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_flat.initialize(m_randSvc, Rndm::Flat(0., 1.)).isFailure()) {
+    error() << "Couldn't initialize RndmGenSvc!" << endmsg;
+    return StatusCode::FAILURE;
+  }
+
+  // check if readout exists
+  if (m_geoSvc->getDetector()->readouts().find(m_readoutName) == m_geoSvc->getDetector()->readouts().end()) {
+    error() << "Readout <<" << m_readoutName << ">> does not exist." << endmsg;
+    return StatusCode::FAILURE;
+  }
+  // set the cellID decoder
+  m_decoder = m_geoSvc->getDetector()->readout(m_readoutName).idSpec().decoder();
+  // retrieve the volume manager
+  m_volman = m_geoSvc->getDetector()->volumeManager();
+
+  return StatusCode::SUCCESS;
+}
+
+StatusCode MUONsimpleDigitizer::execute() {
+  // Get the input collection with Geant4 hits
+  const edm4hep::SimTrackerHitCollection* input_sim_hits = m_input_sim_hits.get();
+  debug() << "Input Sim Hit collection size: " << input_sim_hits->size() << endmsg;
+
+  // Digitize the sim hits
+  edm4hep::TrackerHit3DCollection* output_digi_hits = m_output_digi_hits.createAndPut();
+
+  // Prepare a collection for the association between digitized and simulated hit, setting weights to 1
+  extension::MCRecoMuonSystemDigiAssociationCollection* digi_sim_associations = m_output_sim_digi_association.createAndPut();
+
+  // Prepare collections for the validation distributions
+  auto simDigiDifferenceX = m_simDigiDifferenceX.createAndPut();
+  auto simDigiDifferenceY = m_simDigiDifferenceY.createAndPut();
+  auto simDigiDifferenceZ = m_simDigiDifferenceZ.createAndPut();
+
+  for (const auto& input_sim_hit : *input_sim_hits) {
+    // Apply efficiency
+    if (m_flat.shoot() > m_efficiency) {
+      continue; // Skip this hit
+    }
+
+    auto output_digi_hit = output_digi_hits->create();
+    // smear the hit position:
+    // retrieve the cell detElement
+    dd4hep::DDSegmentation::CellID cellID         = input_sim_hit.getCellID();
+    debug() << "Digitisation of " << m_readoutName << ", cellID: " << cellID << endmsg;
+    //  auto                           cellDetElement = m_volman.lookupDetElement(cellID);
+
+    const auto& stripsTransformMatrix = m_volman.lookupVolumePlacement(cellID).matrix();
```

Comment:
maybe using the segmentation position and cellID functions [[link](https://dd4hep.web.cern.ch/dd4hep/reference/classdd4hep_1_1Segmentation.html#ab894adbb604c9786d2bd801a6c5fac66)] give the same functionality without the overhead of the volume manager?